protostar/commands/deploy/starkware/starknet_cli.py

```diff
@@ -59,25 +58,23 @@ async def deploy(
             )
             break
     else:
-        if len(inputs) > 0:
+        if len(inputs) != 0:
             raise ValueError(
                 "Constructor args cannot be specified for contracts without a constructor."
             )
 
     tx = Deploy(
         contract_address_salt=numeric_salt,
-        contract_definition=contract_definition,
+        contract_definition=contract_class,
         constructor_calldata=inputs,
+        version=constants.TRANSACTION_VERSION,
     )  # type: ignore
 
+    gateway_client = GatewayClient(
+        url=gateway_url, retry_config=RetryConfig(n_retries=1)
+    )
     gateway_response = await gateway_client.add_transaction(tx=tx, token=token)
-
-    if gateway_response["code"] != StarkErrorCode.TRANSACTION_RECEIVED.name:
-        raise DeployContractException(
-            message=f"Failed to send transaction. Response: {gateway_response}"
-        )
-
-    # return gateway_response
```

Comment:
I think we should leave the current code. This kind of errors should be handled by exceptions, not asserts.

Reply:
Agreed